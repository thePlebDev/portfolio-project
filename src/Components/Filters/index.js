import React from 'react';
import styled from 'styled-components'

import useFilter from '../../Hooks/useFilterHook';

const FilterContainer =styled.div`
  display:flex;
  align-items:center;
  justify-content:center;
`


const Text = styled.div`
  margin:5px;
  cursor:pointer;
`

const Filter = ({filters,setBlogState,blogState})=>{

  const {handleClick} = useFilter(setBlogState,blogState)


  return(
    <FilterContainer>
        <Text  onClick={(e)=>handleClick(e)}>{filters}</Text>
    </FilterContainer>
  )
}

export default Filter
