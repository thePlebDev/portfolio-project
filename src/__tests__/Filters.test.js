import React from 'react';

import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';

import Filters from '../Components/Filters'


describe('testing the Filter Component',()=>{
  let wrapper;
  let useMockFunc = jest.fn(()=>{
    return{
      handleClick:jest.fn(()=>{})
    }
  })

  beforeEach(()=>{
    wrapper = shallow(<Filters filters={'it do be like that'} useApiCall={useMockFunc}/>)
  })
  it('should test the snapshot',()=>{
    const tree = renderer.create(<Filters/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should make sure that everything renders properly',()=>{
    expect(wrapper.find('[data-testid="container"]').length).toBe(1)
    expect(wrapper.find('[data-testid="text"]').length).toBe(1)
    expect(wrapper.find('[data-testid="text"]').props().children).toBe('it do be like that')
  })
  it('should test the mock function',()=>{
    expect(useMockFunc.mock.calls.length).toBe(3)
  })
})
