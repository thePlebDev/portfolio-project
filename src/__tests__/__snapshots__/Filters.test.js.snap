// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the Filter Component should test the snapshot 1`] = `
<div
  className="sc-bdnylx eMLUIs"
  data-testid="container"
>
  <div
    className="sc-gtssRu bENsCd"
    data-testid="text"
    onClick={[Function]}
  />
</div>
`;
